modules/ecs-cluster-ec2-asg/v0.0.1/variables.tf

```diff
@@ -0,0 +1,281 @@
+  type        = number
+  default     = 120
+}
+
+variable "alarm_threshold_scale_up" {
+  description = "The value against which the specified statistic is compared."
+  type        = number
+  default     = 100
+}
+
+variable "alarm_threshold_scale_down" {
+  description = "The value against which the specified statistic is compared."
+  type        = number
+  default     = 50
+}
+
+variable "alarm_actions_enabled" {
+  description = "Whether or not to enable alarm actions for scaling up and down."
+  type        = bool
+  default     = true
+}
+
+variable "ssh_key_name" {
+  description = "Name of SSH key pair to use as default (ec2-user) user key"
+  type        = string
+  default     = ""
+}
+
+# ECS related variables
+# Required:
+variable "cluster_name" {
+  description = "Name of the ECS cluster to create"
+  type        = string
+}
+
+variable "container_insights" {
+  description = "Enable or disable Container Insights. Valid values are enabled and disabled"
+  type        = string
+  default     = "disabled"
+  validation {
+    condition     = contains(["enabled", "disabled"], var.container_insights)
+    error_message = "Valid values for container_insights are: enabled, disabled"
+  }
+}
+
+variable "cloudwatch_log_retention" {
+  description = "CloudWatch log group retention in days"
+  type        = number
+  default     = 7
+}
+
+variable "ecs_instance_role_assume_role_policy" {
+  description = "Assume role policy for the instance role"
+  type        = string
+  default     = <<EOF
+{
+  "Version": "2008-10-17",
+  "Statement": [
+    {
+      "Sid": "",
+      "Effect": "Allow",
+      "Principal": {
+        "Service": "ec2.amazonaws.com"
+      },
+      "Action": "sts:AssumeRole"
+    }
+  ]
+}
+EOF
+}
+
+variable "ecs_instance_role_policy" {
+  description = "Instance role policy"
+  type        = string
+  default     = <<EOF
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Effect": "Allow",
+      "Action": [
+        "ecs:CreateCluster",
+        "ecs:DeregisterContainerInstance",
+        "ecs:DiscoverPollEndpoint",
+        "ecs:Poll",
+        "ecs:RegisterContainerInstance",
+        "ecs:StartTelemetrySession",
+        "ecs:Submit*",
+        "ecr:GetAuthorizationToken",
+        "ecr:BatchCheckLayerAvailability",
+        "ecr:GetDownloadUrlForLayer",
+        "ecr:BatchGetImage",
+        "logs:CreateLogStream",
+        "logs:PutLogEvents"
+      ],
+      "Resource": "*"
+    }
+  ]
+}
+EOF
+}
+
+variable "ecs_service_role_assume_role_policy" {
+  description = "Assume role policy for the service role"
+  type        = string
+  default     = <<EOF
+{
+  "Version": "2008-10-17",
+  "Statement": [
+    {
+      "Sid": "",
+      "Effect": "Allow",
+      "Principal": {
+        "Service": "ecs.amazonaws.com"
+      },
+      "Action": "sts:AssumeRole"
+    }
+  ]
+}
+EOF
+}
+
+variable "ecs_service_role_policy" {
```

Comment:
what's the difference between the service role policy and the instance role policy?

Reply:
The service role policy should be for the services that run on the cluster. The instance role policy should be for the EC2 instances that make up the cluster.